daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -169,15 +172,13 @@ object SValue {
       ()
     }
 
-    val Empty = SGenMap(TreeMap.empty)
-
-    def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
+    def apply(isTextMap: Boolean, entries: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
-      SGenMap(xs.map { case p @ (k, _) => comparable(k); p }.to[O])
+      SGenMap(isTextMap, entries.map { case p @ (k, _) => comparable(k); p }.to[O])
     }
 
-    def apply(xs: (SValue, SValue)*): SGenMap =
-      SGenMap(xs.iterator)
+    def apply(isTextMap: Boolean, entries: (SValue, SValue)*): SGenMap =
+      SGenMap(isTextMap: Boolean, entries.iterator)
```

Comment:
Is that even syntactically valid?
```suggestion
      SGenMap(isTextMap, entries.iterator)
```